### PR TITLE
Create 2020 Focus Blog Post

### PR DESCRIPTION
**Pull request by momack2**
TODO: NetflixPost
TODO: "To kick this off, we’ve added XX bounties suggested by our 2020 Theme proposal process, and merged X RFPs or open grant proposals for larger-scale improvements to help benefit the whole community." --> real numbers
TODO: check all the links

### CODE REVIEW

File: content/post/082-our-focus-for-2020.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concepts (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories - allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores. While IPFS had previously taken 24mins to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took 11mins (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: @aschmahmann - can you take a look at this and suggest edits?

File: content/post/082-our-focus-for-2020.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
...r, and landing a ton of features and improvements to make IPFS better for everyone — where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the features and improvements landing in 0.5.0, that future adoption work is significantly unblocked - however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-how adoption cadence, we plan to use channels like DevGrants and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools. Excited to use these new improvements to IPFS in your package management use case? [Make an issue in the package managers repo](https://github.com/ipfs/package-managers/issues) for the latest recommendations!
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: @andrew does this seem right to you as the previous steward of our package managers work? Am I communicating this accurately to our package manager communities?

File: content/post/082-our-focus-for-2020.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+We also formed collaborations with IPFS users to partner on improving IPFS for package management use cases. One of our main collaborations was with [Netflix]() to optimize the speed of fetching large container images with Bitswap, our peer-to-peer data transfer algorithm. You can read more about that specific collaboration - and the resulting 3-5x improvements in Bitswap performance! - in [our previous blog post](TODO:NetflixPost).
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: TODO

File: content/post/082-our-focus-for-2020.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+We also want to grow and support the many amazing contributors to ipfs, and exciting new explorations to build tools or address new use cases with IPFS. There were a ton of amazing suggestions for features and focus areas in our 2020 theme collection that would be immensely valuable for the project, but our core working groups can’t push forward this quarter. To help with that, we’re launching a community DevGrants program where Protocol Labs and other groups can offer bounties, RFPs, and accept open suggestions for improvements, new features, and even new implementations. To kick this off, we’ve added XX bounties suggested by our 2020 Theme proposal process, and merged X RFPs or open grant proposals for larger-scale improvements to help benefit the whole community. If you are a user or contributor to IPFS, this is a great opportunity to both sponsor and apply for [grants](https://github.com/ipfs/devgrants/tree/master/rfps#how-to-create-rfp) or [bounties](https://github.com/ipfs/devgrants/blob/master/BOUNTIES.md#how-to-propose) to help make IPFS better for everyone!
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: TODO: update with fresh #s

File: content/post/082-our-focus-for-2020.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+url: 2020-02-07-our-focus-for-2020
+title: Our Focus for 2020
+author: Molly Mackinlay
+---
+
+_"The mission of IPFS is to create a resilient, upgradeable, open network to preserve and grow humanity’s knowledge."_
+
+It’s a new year, and with it time to update the [IPFS Project Roadmap](https://github.com/ipfs/roadmap) with our focus for 2020. On the IPFS team, we use regular planning cycles to reflect on our progress, zoom out to assess the bigger picture, and realign on the most important work to do next. These cycles help us ensure we stay on track to achieve our mission, and maximize the value and utility we create in the world. 
+
+> Looking for the TL;DR? Hop right over to the [IPFS Roadmap Repo](https://github.com/ipfs/roadmap) to see our 2020 priority and Epics.
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
> Looking for the TL;DR? Hop right over to the [2020 section of our IPFS Roadmap](https://github.com/ipfs/roadmap#2020-priority) to see our focus and epics.
```

File: content/post/082-our-focus-for-2020.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+## Looking back at our 2019 Roadmap
+
+Last year, we did a large 2019 Roadmapping effort to write down our mission, define some of the many long-term goals for the IPFS Project, and prioritize where to focus our energy first. We had a lot of “planning” debt, so going from 0 to 1 on a project roadmap was a large endeavor. Our process involved each working group generating a roadmap around a shared goal, and then merging important work streams into “epics” for the whole project that highlighted our major objectives. You can see how we did against these objectives in our [2019 Project Roadmap](https://github.com/ipfs/roadmap), or dive more into how we chose those goals in our [2019 Roadmap blog post](https://blog.ipfs.io/78-ipfs-2019-roadmap).
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
Last year, we did a large 2019 Roadmapping effort to write down our mission, define some of the many long-term goals for the IPFS Project, and prioritize where to focus our energy first. We had a lot of “planning” debt, so going from 0 to 1 on a project roadmap was a large endeavor. Our process involved each working group generating a roadmap around a shared goal, and then merging important work streams into “epics” for the whole project that highlighted our major objectives. You can see how we did against these objectives in our [2019 Project Roadmap](https://github.com/ipfs/roadmap/blob/master/2019-IPFS-Project-Roadmap.md), or dive more into how we chose those goals in our [2019 Roadmap blog post](https://blog.ipfs.io/78-ipfs-2019-roadmap).
```

File: content/post/082-our-focus-for-2020.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+We also formed collaborations with IPFS users to partner on improving IPFS for package management use cases. One of our main collaborations was with [Netflix]() to optimize the speed of fetching large container images with Bitswap, our peer-to-peer data transfer algorithm. You can read more about that specific collaboration - and the resulting 3-5x improvements in Bitswap performance! - in [our previous blog post](TODO:NetflixPost). 
+
+IPFS Cluster also released [“collaborative clusters”](https://blog.ipfs.io/2020-01-09-collaborative-clusters/), a new feature to enable package manager maintainers and mirrors to add and replicate repositories across a community of IPFS nodes. With collaborative clusters, any maintainer can push new updates to the pinset of data to mirror, which is then sharded and sync’d across all mirroring nodes. We’ve already seen package managers like Pac-Man added to collaborative clusters along with many “data package managers” like Wikipedia and Project Gutenberg - [follow these instructions to add your own!](https://collab.ipfscluster.io/)
+
+## How we did against our goal
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
## How We Did Against our Goal
```

File: content/post/082-our-focus-for-2020.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+- [Content Routing ZenHub Roadmap](https://app.zenhub.com/workspaces/content-routing-2020-5e29be25ab9d8da0b032cfcc/roadmap)
+- [Testground ZenHub Roadmap](https://app.zenhub.com/workspaces/testground-5db6a5bf7ca61c00014e2961/roadmap)
+
+We considered a number of other potential goals - especially all the great [2020 Theme Proposals](https://github.com/ipfs/roadmap/issues/) - before selecting this priority. However, we decided it was more important to focus core working group dev time on the main blockers and pain points to enable the entire ecosystem to grow and succeed. Many of these proposals are actually very well suited for community ownership via [DevGrants](https://github.com/ipfs/devgrants) and collaborations - and some of them, like ["IPFS in Rust"](https://github.com/ipfs/roadmap/issues/54) and ["Examples and Tutorials"](https://github.com/ipfs/roadmap/issues/50), already have grants or bounties associated with them!
+
+## Increasing contributor velocity while supporting adoption
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
## Increasing Contributor Velocity While Supporting Adoption
```

File: content/post/082-our-focus-for-2020.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+To help coordinate this work, we’re creating a new “Ecosystem” working group made up of 3 Special Interest Groups (SIGs) focused on developer experience, collabs and community, and browsers/connectivity. Our Ecosystem working group’s objective is to ensure community health and growth through collaborations, developer experience and platform availability. The 3 SIGs each focus on:
+**Browsers & Connectivity**: Maximize availability & connectivity of IPFS on the web
+**Collabs & Community**: Support IPFS users and grow new opportunities through research, collaborations and community engagement
+**Developer Experience & Maintainership**: Support the IPFS technical community through documentation, contributor experience, API ergonomics and tooling
+
+Curious about the work they’re taking on? Check out the [Project Roadmap Epics](https://github.com/ipfs/roadmap#2020-epics) for more tangible goals, and be on the lookout for more suggestions on how you can help enable other contributors throughout the IPFS ecosystem!
+
+## Everything else
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
## Everything Else
```

File: content/post/082-our-focus-for-2020.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+In the end, we succeeded at staying focused, not building another new package manager, and landing a ton of features and improvements to make IPFS better for everyone; where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the features and improvements landing in 0.5.0, that future adoption work is significantly unblocked, however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-hoc adoption cadence, we plan to use channels like [DevGrants](https://github.com/ipfs/devgrants) and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools.
```
Comment (renrutnnej) [inline content/post/082-our-focus-for-2020.md]: Just to clarify, this is go-ipfs 0.5.0 right?

Comment (momack2) [reply content/post/082-our-focus-for-2020.md]: yep! Should I clarify/link again?

Comment (momack2) [reply content/post/082-our-focus-for-2020.md]:
```suggestion
In the end, we succeeded at staying focused, not building another new package manager, and landing a ton of features and improvements to make IPFS better for everyone; where we didn’t succeed was shipping and integrating those improvements into existing package managers to drive adoption and visibility of IPFS with an awesome community of aligned and talented developers. With the [features and improvements landing in 0.5.0](https://github.com/ipfs/go-ipfs/issues/6776), that future adoption work is significantly unblocked, however we also learned through our research that most package manager communities aren’t fast adopters of new tooling around package distribution due to the semi-pro-bono nature of maintainership. To support a slower, more ad-hoc adoption cadence, we plan to use channels like [DevGrants](https://github.com/ipfs/devgrants) and Collaborations to support package manager adopters acting within their own communities. This allows us to continue focusing our core working groups on improving and augmenting the core protocol and reference implementation, while supporting many community applications and refinements of these tools. 
```

Comment (renrutnnej) [reply content/post/082-our-focus-for-2020.md]: Much better!

File: content/post/082-our-focus-for-2020.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+We also want to grow and support the many amazing contributors to IPFS, and exciting new explorations to build tools or address new use cases. There were a ton of amazing suggestions for features and focus areas in our 2020 theme collection that would be immensely valuable for the project, but our core working groups can’t push forward this quarter. To help with that, we’re launching a community DevGrants program where Protocol Labs and other groups can offer bounties, RFPs, and accept open suggestions for improvements, new features, and even new implementations. 
+
+To kick this off, we’ve added XX bounties suggested by our 2020 Theme proposal process, and merged X RFPs or open grant proposals for larger-scale improvements to help benefit the whole community. If you are a user or contributor to IPFS, this is a great opportunity to both sponsor and apply for [grants](https://github.com/ipfs/devgrants/tree/master/rfps#how-to-create-rfp) or [bounties](https://github.com/ipfs/devgrants/blob/master/BOUNTIES.md#how-to-propose) to help make IPFS better for everyone!
```
Comment (Stebalien) [inline content/post/082-our-focus-for-2020.md]: XX -> TODO (so it's not lost)

File: content/post/082-our-focus-for-2020.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+Our Package Managers goal also helped focus our user research, quarterly objectives, and collaborations. In Q1&2, our Package Managers team surveyed the space and identified core needs documented in our [Package Managers repo](https://github.com/ipfs/package-managers). These insights informed our [quarterly OKR planning](https://github.com/ipfs/team-mgmt/blob/master/OKR/PACKAGE-MANAGERS.md), driving features to make it easier to mirror file system package managers on IPFS. One particular addition was adding metadata on “last updated” time to our unixfs data model to support smarter/faster package updates (humbly named “unixfsv1.5” and already implemented in js-ipfs; coming soon to go). A number of these package-manager-focused improvements are all slated to roll out in our next feature release, go-ipfs 0.5.0 - [feel free to follow along here](https://github.com/ipfs/go-ipfs/issues/6776)!
```
Comment (Stebalien) [inline content/post/082-our-focus-for-2020.md]: We might want to make it clear that none of these changes have been released (e.g., you'll need the latest master to get the add speed improvements).

File: content/post/082-our-focus-for-2020.md (hunk 2 above)
Comment (andrew) [reply content/post/082-our-focus-for-2020.md]: Yeah this is great, nicely sums it up and still leaves the door wide open for future package manager integrations and collabs 👍

Comment (momack2) [reply content/post/082-our-focus-for-2020.md]: Thanks!

File: content/post/082-our-focus-for-2020.md (hunk 11 above)
Comment (Stebalien) [reply content/post/082-our-focus-for-2020.md]: (oh, already marked)

File: content/post/082-our-focus-for-2020.md (hunk 12 above)
Comment (momack2) [reply content/post/082-our-focus-for-2020.md]: added a line for grabbing master to try them out

File: content/post/082-our-focus-for-2020.md (hunk 3 above)
Comment (momack2) [reply content/post/082-our-focus-for-2020.md]: added, but need to be sure to update

File: content/post/082-our-focus-for-2020.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Huge thanks to <a href="https://twitter.com/agentofuser?ref_src=twsrc%5Etfw">@agentofuser</a> <a href="https://twitter.com/dietrich?ref_src=twsrc%5Etfw">@dietrich</a> <a href="https://twitter.com/Ianopolous?ref_src=twsrc%5Etfw">@Ianopolous</a> <a href="https://twitter.com/lidelOrg?ref_src=twsrc%5Etfw">@lidelOrg</a> <a href="https://twitter.com/NukeManDan?ref_src=twsrc%5Etfw">@NukeManDan</a> <a href="https://twitter.com/aschmahmann?ref_src=twsrc%5Etfw">@aschmahmann</a> <a href="https://twitter.com/vasa_develop?ref_src=twsrc%5Etfw">@vasa_develop</a> &amp; <a href="https://twitter.com/hashtag/liyunzhi_666?src=hash&amp;ref_src=twsrc%5Etfw">#liyunzhi_666</a> for the amazing contributions. There’s far more here than one team can build in 2020, but all awesome ideas &amp; much we can empower the whole community to pitch in on! <a href="https://t.co/jUmEJ6nd3k">pic.twitter.com/jUmEJ6nd3k</a></p>&mdash; Molly (@momack28) <a href="https://twitter.com/momack28/status/1193404771983454214?ref_src=twsrc%5Etfw">November 10, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```
Comment (autonome) [inline content/post/082-our-focus-for-2020.md]: 11 proposals noted above, but only 8 ppl listed here - if listing, should list all (including you!)

Comment (momack2) [reply content/post/082-our-focus-for-2020.md]: rust and offline-first came after this tweet (https://twitter.com/momack28/status/1193404771983454214) - I can fake the rewrite here, but I can't change the actual tweet...

Comment (momack2) [reply content/post/082-our-focus-for-2020.md]:
```suggestion
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Huge thanks to <a href="https://twitter.com/agentofuser?ref_src=twsrc%5Etfw">@agentofuser</a> <a href="https://twitter.com/dietrich?ref_src=twsrc%5Etfw">@dietrich</a> <a href="https://twitter.com/Ianopolous?ref_src=twsrc%5Etfw">@Ianopolous</a> <a href="https://twitter.com/lidelOrg?ref_src=twsrc%5Etfw">@lidelOrg</a> <a href="https://twitter.com/NukeManDan?ref_src=twsrc%5Etfw">@NukeManDan</a> <a href="https://twitter.com/aschmahmann?ref_src=twsrc%5Etfw">@aschmahmann</a> <a href="https://twitter.com/vasa_develop?ref_src=twsrc%5Etfw">@vasa_develop</a> <a href="https://twitter.com/hashtag/liyunzhi_666?src=hash&amp;ref_src=twsrc%5Etfw">#liyunzhi_666</a> <a href="https://twitter.com/hashtag/gozala?src=hash&amp;ref_src=twsrc%5Etfw">@gozala</a> &amp; <a href="https://twitter.com/hashtag/parkan?src=hash&amp;ref_src=twsrc%5Etfw">@parkan</a> for the amazing contributions. There’s far more here than one team can build in 2020, but all awesome ideas &amp; much we can empower the whole community to pitch in on! <a href="https://t.co/jUmEJ6nd3k">pic.twitter.com/jUmEJ6nd3k</a></p>&mdash; Molly (@momack28) <a href="https://twitter.com/momack28/status/1193404771983454214?ref_src=twsrc%5Etfw">November 10, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```

File: content/post/082-our-focus-for-2020.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
+---
+date: 2020-02-07
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: update =[

File: content/post/082-our-focus-for-2020.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+We also formed collaborations with IPFS users to partner on improving IPFS for package management use cases. One of our main collaborations was with [Netflix](https://netflix.com) to optimize the speed of fetching large container images with [Bitswap](https://github.com/ipfs/go-bitswap), our peer-to-peer data transfer algorithm. You can read more about that specific collaboration—and the resulting 3-5x improvements in Bitswap performance!—in [our previous blog post](/2020-02-10-improved-bitswap-with-netflix/). 
+
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]: reminder to me to check this link before shipping >.<

File: content/post/082-our-focus-for-2020.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+It’s a new year, and with it time to update the [IPFS Project Roadmap](https://github.com/ipfs/roadmap) with our focus for 2020. As part of that, we also want to reflect on our successes, challenges, and learnings from 2019 to help us stay on track to achieve our mission, and maximize the value and utility we create in the world. 
+
+> Looking for the TL;DR? Hop right over to the [2020 section of our IPFS Roadmap](https://github.com/ipfs/roadmap#2020-priority) to see our focus and epics for the next year.
+
+## Looking Back at 2019
+2019 was a super exciting year for the IPFS Project:
+- The IPFS Public Network grew by **30x in 2019**! 🚀
+- **💯s of thousands of nodes** are participating in the IPFS Network daily
+- **Millions** of users access the IPFS HTTP Gateway every week 🎉
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
- **Millions of users** access the IPFS HTTP Gateway every week 🎉
```

File: content/post/082-our-focus-for-2020.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+- There are **hundreds of dapps, tools, and projects** in the wider IPFS Ecosystem - including newcomers like [Anytype](https://anytype.io/), [Microsoft ION](https://techcommunity.microsoft.com/t5/azure-active-directory-identity/toward-scalable-decentralized-identifier-systems/ba-p/560168), [Haven](https://gethaven.app/) by OB1, [Brave](https://brave.com/), [3box](https://3box.io/), [EthDNS](https://medium.com/the-ethereum-name-service/ethdns-9d56298fa38a) and more!! 🙌
+
+![IPFS Ecosystem Diagram](https://user-images.githubusercontent.com/618519/74208137-0fc98a00-4c37-11ea-800b-ebd61485d332.png)
+See our [Ecosystem Update from IPFS Camp](https://www.youtube.com/watch?v=jpQnQbfhuBc&feature=youtu.be) for more!
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
```

File: content/post/082-our-focus-for-2020.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+
+![IPFS Ecosystem Diagram](https://user-images.githubusercontent.com/618519/74208137-0fc98a00-4c37-11ea-800b-ebd61485d332.png)
+See our [Ecosystem Update from IPFS Camp](https://www.youtube.com/watch?v=jpQnQbfhuBc&feature=youtu.be) for more! 
+
+This growth caused us to shift some of our attention midyear to support new usage and demand - re-focusing chunks of our team on improving documentation, gateway performance, and our testing tools to validate large-scale network upgrades (see our [Operation Task-Force blog post](https://blog.ipfs.io/2019-07-31-operation-task-force/) for details). We still made progress on our Package Managers goal, but less than we hoped given also investing heavily in other critical areas to support the ecosystem. 🤞
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
This growth caused us to shift some of our attention midyear to support new usage and demand - re-focusing some of our working groups on improving documentation, gateway performance, and our testing tools to validate large-scale network upgrades (see our [Operation Task-Force blog post](https://blog.ipfs.io/2019-07-31-operation-task-force/) for details). We still made progress on our Package Managers goal, but less than we hoped given **also** investing heavily in other critical areas to support the ecosystem. 🤞
```

File: content/post/082-our-focus-for-2020.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concept (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories—allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores. While IPFS had previously taken **24mins** to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took **11mins** (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_
```
Comment (momack2) [inline content/post/082-our-focus-for-2020.md]:
```suggestion
Focusing on a representative use case like package managers brought focus and structure to our prioritization of improvements to IPFS. We built a number of proofs-of-concept (POCs) like [apt-on-ipfs](https://github.com/ipfs-shipyard/apt-on-ipfs), [npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), [clojars-on-ipfs](https://github.com/ipfs-shipyard/clojars-mirror-test) and [homebrew-on-ipfs](https://github.com/ipfs/package-managers/issues/12) to analyze how well IPFS performed against user expectations for adding, updating, and fetching large package repositories—allowing us to identify and fix major pain points. For example, our POCs identified a huge bottleneck in adding GBs of data to IPFS. Further testing resulted in _doubling_ the speed of adding on Linux & OSX devices by switching to async-by-default datastores.🎉 While IPFS had previously taken **24mins** to add an Arch Linux repo, after [these fixes](https://github.com/ipfs/go-ipfs/issues/6523#issuecomment-546822191) it only took **11mins** (comparable to the time to copy/paste)! _Note: improvement seen on the badger datastore, already 3x faster than flatfs!_ 
```

**Pull request merged**